index.js

```diff
@@ -149,12 +151,16 @@ class HtmlWebpackPlugin {
           compilation.errors.push(prettyError(templateResult.error, compiler.context).toString());
         }
 
-        const childCompilationOutputName = compilation.mainTemplate.getAssetPath(this.options.filename, 'compiledEntry' in templateResult ? {
+        const compiledEntries = 'compiledEntry' in templateResult ? {
           hash: templateResult.compiledEntry.hash,
           chunk: templateResult.compiledEntry.entry
         } : {
           hash: templateResult.mainCompilationHash
-        });
+        };
+
+        const childCompilationOutputName = webpackMajorVersion === 5
```

Comment:
I would invert the logic to support future versions too (and as it is done in other parts of the codebase)

```
= webpackMajorVersion === 4 ? ... : ...
```

Reply:
That was my first thought too, but we'll lose backward compatibility with v3 and lower, and I'm not 100% that we should support them also? And also this `Compilation` syntax is different only in webpack v5.

Reply:
> That was my first thought too, but we'll lose backward compatibility with v3 and lower, and I'm not 100% that we should support them also? And also this `Compilation` syntax is different only in webpack v5.

Compatibility is stated by:

```
  "peerDependencies": {
    "webpack": ">=4.0.0 < 6.0.0"
  },
```

https://github.com/jantimon/html-webpack-plugin/blob/115bd8a5f8641dc2bd58dd5efefe360e3cdc0102/package.json#L64

Reply:
I've one more idea to pick major `v5` and higher like:
```
webpackMajorVersion >= 5 ? ... : ...
```
What do you think?

Reply:
I was more looking to how it was written in `file-watcher-api.js`.

https://github.com/jantimon/html-webpack-plugin/blob/115bd8a5f8641dc2bd58dd5efefe360e3cdc0102/lib/file-watcher-api.js#L11